.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,39 @@
+name: Build and Test Gem 
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
I'd be careful about macos runners since they are 10x more expensive, and we only have 2000 minutes per month https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#minute-multipliers

Reply:
Maybe we could run only Windows and ubuntu for PRs, and add MacOS for version releases 🤝